plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/base/impl/FeatureAttribute.java

```diff
@@ -72,35 +60,43 @@ public FeatureAttribute() {
 	public FeatureAttribute(String name, String value, String type, String unit, boolean recursive, boolean configurable) {
 		this.name = name;
 		this.value = value;
-		setTypeFromString(type);
+
+		type = type.toUpperCase();
+		for (final Types typeNames : Types.values()) {
+			if (typeNames.toString().equals(type)) {
+				this.type = typeNames;
+			}
+		}
+
 		this.unit = unit;
 		this.recursive = recursive;
 		this.configurable = configurable;
+
 	}
 
 	/**
 	 * @return True, if the value of this FeatureAttribute matches its type. Else false.
 	 */
 	public boolean checkValue() {
-		if (type.toString().equals(LONG)) {
-			try {
-				Long.parseLong(value);
-				return true;
-			} catch (final NumberFormatException e) {
-				return false;
+		if (!value.isEmpty()) {
+			if (type.toString().equals(LONG)) {
+				try {
+					Long.parseLong(value);
+				} catch (final NumberFormatException e) {
+					return false;
+				}
 			}
-		}
-		if (type.toString().equals(DOUBLE)) {
-			try {
-				Double.parseDouble(value);
-				return true;
-			} catch (final NumberFormatException e) {
-				return false;
+			if (type.toString().equals(DOUBLE)) {
+				try {
+					Double.parseDouble(value);
+				} catch (final NumberFormatException e) {
+					return false;
+				}
 			}
-		}
-		if (type.toString().equals(BOOLEAN)) {
-			if (value.toLowerCase().equals("true") || value.toLowerCase().equals("false")) {
-				return true;
+			if (type.toString().equals(BOOLEAN)) {
+				if (value.toLowerCase().equals("true") || value.toLowerCase().equals("false")) {
+					return true;
+				}
 			}
```

Comment:
Here there is an error that was already solved by a pull request, but now it isn't. Please fix.